### ISSUE TITLE
Stuck on exit vehicle

### ISSUE THREAD

**Issue by TuK4z** [altmp/altv-issues#1772]
### Description of the problem

When remove weapons while you are still in vehicle player stuck on exit.
Strange that you have to hold W (go forward) button to get this behavior and it's happening not for every type of vehicle. In this example i use Asbo.

https://streamable.com/r6v2hx

Only tested on C#

### Reproduction steps

[Command("gw")]
        public void gw(IPlayer player)
        {
            player.GiveWeapon(0x13532244, 100, true);
        }
        [Command("rw")]
        public void rw(IPlayer player)
        {
            player.RemoveAllWeapons();
        }
     
Give yourself weapon and remove while in vehicle.
Exit vehicle and hold W. It works just on specific vehicles for example "Asbo".

### Expected behaviour

Not to stuck on exit

### Operating system

Win 11 and Linux 22.04

### Version

13.48 (release)

### Reproduction tested

- [X] I confirm that I have made sure that this issue is also present on the newest dev version

**Comment by litolax** [comment]
Can not confirmed on the latest dev (14.0-dev57)

**Comment by TuK4z** [comment]
I did not find a way to do it on empty server, i guess then it's resource or some data problem i dunno...

**Comment by BluePeer** [comment]
this is a fundamental problem not of the code but of the developer of the resource.
certain events are transmitted, or in other words, ensure that both the server and the client transmit what I call a complete player state.
part of the player state in the vehicle is a value that takes care of the piece outside of the vehicle, there of course completely wrong but wanted in the vehicle.

if you trigger the exit, the status in the client changes from "stuck" to "free" in this chain of events, but if you change weapons (which you should never allow)
then the weapon change event transfers the old "stuck" status as the current "status" and thus overwrites the "free" to "stuck"

Of course, this only happens if the resource has events that take care of it.

as an example like here

if you are in an async resource using Task
then getting out causes the char to start the chain of actions.
event chain 1000ms
- get off at 50ms <-> at 200ms state change <-> 500ms player outside vehicle
-at 190ms weapon change event with old state -> at 250ms transfer of weapon change with old state to client

Result player with wrong state outside of vehicle


only to explain to the opener why he has the problem, where it arises and how he can work around it.
this is to be observed in principle when status changes act in parallel (when weapon changes, vehicle changes, life/dead changes, and similar things happen at the same time or within "server <-> client" "ping pong")

**Comment by TuK4z** [comment]
Found a problem of all this situation. It's all because in clientside i use Config flag of weapon swap.

Alt.SetConfigFlag("DISABLE_AUTO_WEAPON_SWAP", true);

It's possible to fix or it's just how it works?


Reproduction would be like at the top explained with give weapon and remove weapon while in vehicle but in clientside you have to set config flag DISABLE_AUTO_WEAPON_SWAP when you join server.

**Comment by yannbcf** [comment]
This is a bug but fixing it would require deep researches and debugging of why it happens. Tho, I see few workarounds already to fix that issue as the server code level.

You can for example listen for the clientside taskChange event and look for the task id leaving vehicle which get triggered when you press F when in the vehicle, disable the DISABLE_AUTO_WEAPON_SWAP flag and enable it back when this task ends.

**Comment by TuK4z** [comment]
Well i spent about 15 hours while searching why it's happening :D so yeah pain in the a*s stuff like this. But yep you are right i could do that work around. I ll check.
But i think issue could be open then, maybe some poor souls will face same issue and find how to fix it faster or issue will be some day solved it self.

**Comment by yannbcf** [comment]
The issue should stay open as it's a bug with an api alt:V provide.

**Comment by PlumeDreams** [comment]
Yes, I never found out exactly what caused this until I saw this issue
